packages/cli/src/zosfiles/compare/CompareBaseHelper.ts

```diff
@@ -0,0 +1,230 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { IHandlerParameters, DiffUtils } from "@zowe/imperative";
+import {  IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
+
+/**
+ * Base helper for compare command to handle common operations through the comparison
+ * @export
+ */
+export default class ComapareBaseHelper {
+
+    /**
+     * Singleton instance of this class
+     * @private
+     * @static
+     * @type {ComapareBaseHelper}
+     * @memberof ComapareBaseHelper
+     */
+    private static hInstance: ComapareBaseHelper;
+
+    /**
+     * Return a singleton instance of this class
+     * @static
+     * @readonly
+     */
+    public static get instance(): ComapareBaseHelper {
+        if(this.hInstance == null){
+            this.hInstance = new ComapareBaseHelper();
+        }
+
+        return this.hInstance;
+    }
+
+    /**
+     * binary option for file1
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public binary: boolean;
+
+    /**
+     * encoding option for file1
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public encoding: string;
+
+    /**
+     * record option for file1
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public record: boolean;
+
+    /**
+     * volume serial option for file
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public volumeSerial: string;
+
+    /**
+     * responseTimeout option
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public responseTimeout: number;
+
+    /**
+     * binary option for file2
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public binary2: boolean;
+
+    /**
+     * encoding option for file1
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public encoding2: string;
+
+    /**
+     * record option for file2
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public record2: boolean;
+
+    /**
+     * volume serial option for file2
+     * @public
+     * @memberof ComapareBaseHelper
+     */
+    public volumeSerial2: string;
```

Comment:
Instead of defining each option twice (`binary`, `binary2`, `encoding`, `encoding2`, etc), could we create an interface `ICompareFileOptions`? This could remove even more code duplication 🙂 

For example, with an interface like the following:
```
export interface ICompareFileOptions {
    binary: boolean;
    encoding: string;
    record: boolean;
    volumeSerial: string;
}
```

The options could be defined in this class with just 2 lines:
```
public file1Options: ICompareFileOptions;

public file2Options: ICompareFileOptions;
```